actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -77,7 +95,7 @@ class ActionCable.Connection
       {identifier, message, type} = JSON.parse(event.data)
       switch type
         when message_types.welcome
-          @monitor.recordConnect()
+          break
```

Comment:
Your comment got me thinking about terminology, and I guess the question is: what determines a successful connection? And I guess the whole point of this PR is that we regard a successful connection as one where we have a socket connection _and_ we have a successful protocol negotiation. So I put the `recordConnect` back where it was, and also moved the subscription `reload` there as well, because that was a race condition/bound to break.